### ISSUE TITLE
[DCR] [GDPR support] Add a DELETE-ME step in composer to allow user to delete all state data

### ISSUE THREAD

**Issue by p-nagpal** [microsoft/botframework-sdk#5494]
## Issue
To support GDPR, a bot developer needs to allow their users to delete any state/memory collected during a conversation (or user profile data). A declarative step can help them do this as part of the dialog flow and properly reset the state of the dialog.

## Proposed change
Add a DELETE-ME step in Composer to allow bot developer to enable users to delete all state data for the current conversation. 

## Component Impact 
*Describe which components need to be updated*

## Customer Impact 
*Describe the impact on customers*

## Tracking Status
**Dotnet SDK**
- [ ] PR
- [ ] Merged

**Javascript SDK**
- [ ] PR
- [ ] Merged

**Java SDK**
- [ ] PR
- [ ] Merged

**Python SDK**
- [ ] PR
- [ ] Merged

**Emulator**
- [ ] PR
- [ ] Merged

**Samples**
- [ ] PR
- [ ] Merged

**Docs**
- [ ] PR
- [ ] Merged

**Tools**
- [ ] PR
- [ ] Merged

[dcr]

**Comment by Stevenic** [comment]
There are a couple of work items that would be needed to enable this:

- We need to add a new interface that Storage providers can implement to let us tell them to delete everything with a given UserID.   This would instruct the user to search for all records related to a user and delete those records. **Is delete enough or do we also need export?**
- We need to add a new `DeleteUser` step that the bot can call to delete a given user and all of their data.  We should probably also add a `DeleteConversation` step while we're at it...

**Comment by p-nagpal** [comment]
@Stevenic , you are right, we need an export option. If the bot developer would like to retain user data for longer term then an export option is required in addition to delete.

**Comment by gabog** [comment]
Hi @vishwacsena , can we close this one?

**Comment by dinobrago** [comment]
Has this been implemented?
